server-spi/src/main/java/org/keycloak/storage/client/ClientLookupProvider.java

```diff
@@ -71,22 +73,38 @@
      * @param clientId Searched substring of the public client
      *   identifier ({@code client_id} in OIDC or {@code entityID} in SAML.)
      * @param firstResult First result to return. Ignored if negative or {@code null}.
-     * @param maxResults Maximim number of results to return. Ignored if negative or {@code null}.
-     * @return Model of the client, or {@code null} if no client is found.
+     * @param maxResults Maximum number of results to return. Ignored if negative or {@code null}.
+     * @return List of ClientModel or an empty list if no client is found.
+     * @deprecated Use {@link #searchClientsByClientIdStream(org.keycloak.models.RealmModel, java.lang.String, java.lang.Integer, java.lang.Integer)} instead.
      */
-    List<ClientModel> searchClientsByClientId(RealmModel realm, String clientId, Integer firstResult, Integer maxResults);
+    @Deprecated
+    default List<ClientModel> searchClientsByClientId(RealmModel realm, String clientId, Integer firstResult, Integer maxResults) {
```

Comment:
This method wasn't declared in previous version of Keycloak, so I believe we don't have to deprecate it and we can remove it instead.

Reply:
removed